lib/Runtime/Library/JavascriptGenerator.cpp

```diff
@@ -171,6 +172,18 @@ Var JavascriptGenerator::CallGenerator(Var data, ResumeYieldKind resumeKind)
     JavascriptLibrary* library = scriptContext->GetLibrary();
     Var result = nullptr;
 
+    if (this->frame)
+    {
+        // if the function already has a state it may be going to resume in the jit
+        // if so copy any innerScopes into registers jit can access
+        uint32 innerScopeCount = this->scriptFunction->GetFunctionBody()->GetInnerScopeCount();
+        for (uint32 i = 0; i < innerScopeCount; ++i)
+        {
+            Js::RegSlot reg = this->scriptFunction->GetFunctionBody()->GetFirstInnerScopeRegister() + i;
+            this->frame->SetNonVarReg(reg, this->frame->InnerScopeFromIndex(i));
+        }
```

Comment:
This is effectively an inverse of lines 1742-1748 of BailOut.cpp.

NOTE - getting rid of the logic in BailOut.cpp would not be a sufficient fix as it's possible for a generator to start in the interpreter and transition to the Jit at a yield point